### PR TITLE
Added patchable model Id field.

### PR DESCRIPTION
**Pull request by olaaun**
WIP

### CODE REVIEW

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/noderepository/NodeRepositoryNode.java [hunk 1]
```diff
@@ -417,6 +427,7 @@ public String toString() {
                ", description='" + description + '\'' +
                ", history=" + Arrays.toString(history) +
                ", allowedToBeDown=" + allowedToBeDown +
+                ",modelId=" + modelId +
```
Comment (andreer) [inline controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/noderepository/NodeRepositoryNode.java]: needs more space?
```suggestion
                ", modelId=" + modelId +
```

**Review by andreer — APPROVED** [review]
LGTM so far.

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/Node.java [hunk 2]
```diff
@@ -98,6 +99,7 @@ public Node(String id, Set<String> ipAddresses, Set<String> ipAddressPool, Strin
         this.history = history;
         this.type = type;
         this.reports = reports;
+        this.modelId = modelId;
```
Comment (mpolden) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/Node.java]: Should check non-null contract like the other fields.

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/noderepository/bindings/NodeRepositoryNode.java [hunk 3]
```diff
@@ -54,6 +54,8 @@
     public Long currentFirmwareCheck;
     @JsonProperty("wantedFirmwareCheck")
     public Long wantedFirmwareCheck;
+    @JsonProperty("ModelId")
```
Comment (mpolden) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/noderepository/bindings/NodeRepositoryNode.java]: Shouldn't this be `modelId`?

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/Node.java [hunk 4]
```diff
@@ -262,6 +270,8 @@ private Allocation requireAllocation(String message) {
     /** Returns all the reports on this node. */
     public Reports reports() { return reports; }
 
+    public Optional<String> modelId() { return modelId; }
```
Comment (mpolden) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/Node.java]: Please add javadoc explaining what this is as the field name is not very descriptive on its own.

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/persistence/NodeSerializer.java [hunk 5]
```diff
@@ -123,6 +124,7 @@ private void toSlime(Node node, Cursor object) {
         node.status().osVersion().ifPresent(version -> object.setString(osVersionKey, version.toString()));
         node.status().firmwareVerifiedAt().ifPresent(instant -> object.setLong(firmwareCheckKey, instant.toEpochMilli()));
         node.reports().toSlime(object, reportsKey);
+        node.modelId().ifPresent(modelId -> object.setString(modelIdKey, modelId));
```
Comment (mpolden) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/persistence/NodeSerializer.java]: Please update `SerializationTest` to cover (de)serialization of this particular field.

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/Node.java (hunk 4 above)
Comment (mpolden) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/Node.java]: Should probably expose this field in `NodesResponse` as well?

**Pull request merged**